ksqldb-common/src/main/java/io/confluent/ksql/schema/ksql/SystemColumns.java

```diff
@@ -57,35 +60,32 @@
       WINDOWEND_NAME
   );
 
-  private static final Set<ColumnName> PSEUDO_COLUMN_VERSION_ZERO_NAMES
-      = ImmutableSet.<ColumnName>builder()
-      .add(ROWTIME_NAME)
-      .build();
-
-  private static final Set<ColumnName> PSEUDO_COLUMN_VERSION_ONE_NAMES
-      = ImmutableSet.<ColumnName>builder()
-      .addAll(PSEUDO_COLUMN_VERSION_ZERO_NAMES)
-      .add(ROWPARTITION_NAME)
-      .add(ROWOFFSET_NAME)
-      .build();
-
-  private static final Map<Integer, Set<ColumnName>> PSEUDO_COLUMN_NAMES_BY_VERSION =
-      ImmutableMap.of(
-          0, PSEUDO_COLUMN_VERSION_ZERO_NAMES,
-          1, PSEUDO_COLUMN_VERSION_ONE_NAMES
-      );
-
-  private static final Set<ColumnName> MUST_BE_MATERIALIZED_FOR_TABLE_JOINS =
-      ImmutableSet.of(
-          ROWPARTITION_NAME,
-          ROWOFFSET_NAME
-      );
-
-  private static final Map<Integer, Set<ColumnName>> SYSTEM_COLUMN_NAMES_BY_VERSION =
-      ImmutableMap.of(
-          0, buildColumns(0),
-          1, buildColumns(1)
-      );
+  private static final PseudoColumn ROWTIME = PseudoColumn.of(
+      ROWTIME_NAME,
+      ROWTIME_TYPE,
+      ROWTIME_PSEUDOCOLUMN_VERSION,
+      false
+  );
+
+  private static final PseudoColumn ROWPARTITION = PseudoColumn.of(
+      ROWPARTITION_NAME,
+      ROWPARTITION_TYPE,
+      ROWPARTITION_ROWOFFSET_PSEUDOCOLUMN_VERSION,
+      true
+  );
+
+  private static final PseudoColumn ROWOFFSET = PseudoColumn.of(
+      ROWOFFSET_NAME,
+      ROWOFFSET_TYPE,
+      ROWPARTITION_ROWOFFSET_PSEUDOCOLUMN_VERSION,
+      true
+  );
```

Comment:
nit: inline these into the creation of `pseudoColumn` below. They are not used elsewhere and the code is slightly more readable by having fewer static variables IMO.

Reply:
I was debating this, if you think it's good then I'm all for this change. I trust a reader to judge readability over a writer :)